### PR TITLE
chore: xcm benchmarks

### PR DESCRIPTION
**Pull request by sander2**
The xcm benchmarking situation is a bit of a disaster.. This pr adds benchmarks for _most_ of the xcm instructions. 

- A couple instructions are missing, since the benchmark does not include them (I opened an [issue](https://github.com/paritytech/polkadot/issues/7167)).. Kusama uses [hardcoded](https://github.com/paritytech/polkadot/blob/689623c97071d9fac951a459f0b5100b34697a62/runtime/kusama/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs#LL74C40-L74C40) weights for these instructions. 
- As discussed on discord, we're not actually changing the weights right now, since preferably we'd avoid breaking partner's ui's. Tests are added to check that the measured weights are less than our fixed weight. Those tests are breaking for me since we need to run the benchmarks on production hardware, so I disabled them for the time being. But with the number I got, I really doubt that our existing weight is sufficient in the worst-case (since some instructions can operate on multiple assets, so they have to iterate).
- Kusama xcm weights have 0 for the proof sizes. Possibly because they haven't run the benchmarks in a long time. The tests that I added also ignores the proof size component.

### CODE REVIEW

File: parachain/runtime/interlay/src/xcm_config.rs [hunk 1]
```diff
@@ -430,3 +434,235 @@ impl orml_xtokens::Config for Runtime {
...
+            //         );
+            //         // set 1-1
+            //         <AssetManager as xcm_primitives::UnitsToWeightRatio<
+            //             <Runtime as PalletAssetManagerConfig>::ForeignAssetType>
+            //         >::set_units_per_second(
+            //             location.clone().into(),
+            //             1_000_000_000_000u128
+            //         );
+            //     }
+            // }
```
Comment (gregdhill) [inline parachain/runtime/interlay/src/xcm_config.rs:538]: Remove

File: parachain/src/chain_spec/kintsugi.rs [hunk 2]
```diff
@@ -255,7 +255,7 @@ fn kintsugi_mainnet_genesis(
             inflation: FixedU128::checked_from_rational(2, 100).unwrap(), // 2%
         },
         polkadot_xcm: kintsugi_runtime::PolkadotXcmConfig {
-            safe_xcm_version: Some(2),
+            safe_xcm_version: Some(3),
```
Comment (gregdhill) [inline parachain/src/chain_spec/kintsugi.rs:258]: Is this a noteworthy change? Why is it only now required?

File: parachain/runtime/kintsugi/src/xcm_config.rs [hunk 3]
```diff
@@ -424,3 +428,235 @@ impl orml_xtokens::Config for Runtime {
...
+            //         );
+            //         // set 1-1
+            //         <AssetManager as xcm_primitives::UnitsToWeightRatio<
+            //             <Runtime as PalletAssetManagerConfig>::ForeignAssetType>
+            //         >::set_units_per_second(
+            //             location.clone().into(),
+            //             1_000_000_000_000u128
+            //         );
+            //     }
+            // }
```
Comment (gregdhill) [inline parachain/runtime/kintsugi/src/xcm_config.rs:532]: Remove

File: parachain/src/chain_spec/kintsugi.rs (hunk 2 above)
Comment (sander2) [reply parachain/src/chain_spec/kintsugi.rs:258]: It's required for the benchmarks. Since this is part of genesis it doesn't affect mainnet

**Pull request merged**